### DISCUSSION TITLE
QPACK发送Section Acknowledgment的逻辑

### DISCUSSION THREAD

**Discussion by tianxinheihei** [alibaba/xquic#296] [General]
hi，线下测试发现xquic在收到并解析请求后没有发送Section Acknowledgment，看了下代码，是有这段逻辑的，但是却不会触发，是我对设计理解有误么？
如下图所示，xqc_qpack.c代码中虽然有发送Section Acknowledgment的逻辑，但是fin是XQC_FALSE，因此并不会触发Section Acknowledgment的发送逻辑。

在解析http3 headers frame后，pctx->state == XQC_H3_FRM_STATE_PAYLOAD
xqc_bool_t fin = pctx->state == XQC_H3_FRM_STATE_END ? XQC_TRUE : XQC_FALSE;

<img width="686" alt="image" src="https://user-images.githubusercontent.com/27761212/225833388-f0f99806-f2f7-4bd1-b7d7-8e483e2cb100.png">

**Comment by Kulsk** [comment]
我估计是你的HEADERS里边，所有的header能够在静态表找到，或者动态表里边已经包含了这些条目。

代码中这个fin是表示解析完一个HEADERS帧，xqc_rep_get_ric是表示是否有required insert count，所以只有解析完header，并且当前header里边，需要新的动态表条目，才会返回SA，相当于说告诉编码侧收到了动态表条目，并且已经成功解码HEADERS帧。


这个Section Acknowledgment是跟Encoded Field Section相对应的：
![image](https://github.com/alibaba/xquic/assets/22413114/b40eca10-3a6b-47a4-b606-7aa9d15c283c)

![image](https://github.com/alibaba/xquic/assets/22413114/a785d427-e101-48f0-917e-a21e5c55c42b)
